bot/utils/helpers.py

```diff
@@ -37,5 +37,8 @@ def join_role_stats(role_ids: List[int], name: str, guild: Guild) -> Dict[str, i
     """Return a dict object with the number of `members` of each role given, and the `name` for this joined group."""
     members = []
     for role_id in role_ids:
-        members += guild.get_role(role_id).members
+        if (role := guild.get_role(role_id)) is None:
+            raise BadArgument("Unable to fetch role data, the specified role does not exist.")
```

Comment:
Error should be more specific. If this were to go off, neither the user or anyone looking at the logs would exactly be able to identify which role is missing.
```suggestion
            raise ValueError(f"Could not fetch data for role {role} for server embed.")
```

Reply:
Looking at it now, the code in `get_member_counts` that runs before the call to this function doesn't have any error handling and will raise an attribute error for non existing roles. 
We should make that consistent with not handling the same error (for different roles) here, raising a common error for both cases, or providing a default so the command can be used in servers that aren't completely configured

Reply:
I added a custom error to handle both cases!